.github/scripts/submit-provider-key.sh

```diff
@@ -94,4 +96,26 @@ git push -u origin "${branch}"
 # Create pull request and update issue
 pr=$(gh pr create --title "${TITLE}" --body "${msg} ${keyfile/.././} for provider ${namespace}. Closes #${NUMBER}.") #--assignee opentofu/core-engineers)
 gh issue comment "${NUMBER}" -b "Your submission has been validated and has moved on to the pull request phase (${pr}).  This issue has been locked."
-gh issue lock "${NUMBER}" -r resolved
\ No newline at end of file
+gh issue lock "${NUMBER}" -r resolved
+
+if [[ -n "${repo}" ]]; then
+  apt update && apt install -y gpg
+  # get the latest release of the provider
+  latest_release=$(gh release list --exclude-drafts --exclude-pre-releases --repo "${repo}" -L 1 -O desc --json name -q '.[].name')
```

Comment:
I wonder if we need to do it against the latest release, or just one of the releases?

I'm concerned someone could be uploading a key for an older version of a provider.

Maybe we just make this a warning and say that it should be manually checked by a human?

Reply:
> I wonder if we need to do it against the latest release, or just one of the releases? I'm concerned someone could be uploading a key for an older version of a provider.
You're right, but
> Maybe we just make this a warning and say that it should be manually checked by a human?
I wanted to have this more of a warning, indeed.
Since we (or at least I) run those checks manually, I will do it manually when I see that warning:
```shell
  if ! gpg --verify "${sigfile}" "${shafile}"
  then
    gh issue comment "${NUMBER}" -b "Failed to validate the submitted key against ${repo}@${latest_release}. Could this key be for an older version of the provider?"
  fi
```